packages/blade/src/components/Carousel/__tests__/Carousel.web.test.tsx

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+/* eslint-disable @typescript-eslint/require-await */
+import { fireEvent, act } from '@testing-library/react';
+import { Carousel } from '../Carousel';
+import { CarouselItem } from '../CarouselItem';
+import { useCarouselContext } from '../CarouselContext';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+
+const TestimonialCard = ({ index }: { index: number }): React.ReactElement => {
+  const { activeSlide } = useCarouselContext();
+  return (
+    <div>
+      <p>{index}</p>
+      {/* A hack to get the internal state of the carousel so that i can assert the tests */}
+      {/* Users won't be able to do this since useCarouselContext is not gonna be exported */}
+      <p data-testid="active-slide">{activeSlide}</p>
+    </div>
+  );
+};
+
+beforeAll(() => {
+  Object.defineProperty(window.Element.prototype, 'scroll', {
+    writable: true,
+    value: jest.fn(),
+  });
+
+  Object.defineProperty(window.Element.prototype, 'scrollLeft', {
+    writable: true,
+    value: jest.fn(),
+  });
+
+  Object.defineProperty(window.Element.prototype, 'getBoundingClientRect', {
+    writable: true,
+    value: jest.fn(() => ({ width: 100, left: 0 })),
+  });
+});
+
+describe('<Carousel />', () => {
+  it('should go to next/previous slide', () => {
+    const onChange = jest.fn();
+    const { getByRole, queryAllByRole, queryAllByTestId } = renderWithTheme(
+      <Carousel visibleItems={1} onChange={onChange}>
+        <CarouselItem>
+          <TestimonialCard index={0} />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard index={1} />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard index={1} />
+        </CarouselItem>
+        <CarouselItem>
+          <TestimonialCard index={1} />
+        </CarouselItem>
+      </Carousel>,
+    );
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+
+    const nextButton = getByRole('button', { name: 'Next Slide' });
+    const previousButton = getByRole('button', { name: 'Previous Slide' });
+    fireEvent.click(nextButton);
+    fireEvent.click(nextButton);
+
+    expect(nextButton).toBeInTheDocument();
+    expect(previousButton).toBeInTheDocument();
+
+    expect(onChange).toHaveBeenLastCalledWith(2);
```

Comment:
What does `2` signifies here? 3rd slide?

Reply:
Maybe add a more clear assetions, is something like the below possible?Asserting on text on current active slide
```suggestion
    expect(onChange).toHaveBeenLastCalledWith(2);
    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('2');
```

Reply:
Yes, 3rd slide. 

Avoiding to use `data-testid="active-slide"` many times since it's kind of using a hack by getting the internal state via useCarouselContext() in TestimonialCard. And is testing the implementation detail. 

onChange is better here since it maps well with how a user will use this.